From 12936db649e7c1884037b93ff612cce732e05cf1 Mon Sep 17 00:00:00 2001
From: William Woodall <wjwwood@gmail.com>
Date: Thu, 31 Jan 2013 23:47:06 -0800
Subject: [PATCH] Custom rules file

---
 debian/rules |   89 ++++++++++++++++++++++++----------------------------------
 1 file changed, 37 insertions(+), 52 deletions(-)
 rewrite debian/rules (78%)

diff --git a/debian/rules b/debian/rules
dissimilarity index 78%
index 9a59446..7ce32c0 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1,52 +1,37 @@
-#!/usr/bin/make -f
-# -*- makefile -*-
-# Sample debian/rules that uses debhelper.
-# This file was originally written by Joey Hess and Craig Small.
-# As a special exception, when this file is copied by dh-make into a
-# dh-make output file, you may use that output file without restriction.
-# This special exception was added by Craig Small in version 0.37 of dh-make.
-
-# Uncomment this to turn on verbose mode.
-export DH_VERBOSE=1
-export DH_OPTIONS=-v --buildsystem=cmake
-# TODO: remove the LDFLAGS override.  It's here to avoid esoteric problems
-# of this sort:
-#  https://code.ros.org/trac/ros/ticket/2977
-#  https://code.ros.org/trac/ros/ticket/3842
-export LDFLAGS=
-export PKG_CONFIG_PATH=/opt/ros/fuerte/lib/pkgconfig
-
-%:
-	dh  $@
-
-override_dh_auto_configure:
-	# In case we're installing to a non-standard location, look for a setup.sh
-	# in the install tree that was dropped by catkin, and source it.  It will
-	# set things like CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
-	if [ -f "/opt/ros/fuerte/setup.sh" ]; then . "/opt/ros/fuerte/setup.sh"; fi && \
-	dh_auto_configure -- \
-		-DCATKIN_BUILD_BINARY_PACKAGE="1" \
-		-DCMAKE_INSTALL_PREFIX="/opt/ros/fuerte" \
-		-DCMAKE_PREFIX_PATH="/opt/ros/fuerte"
-
-override_dh_auto_build:
-	# In case we're installing to a non-standard location, look for a setup.sh
-	# in the install tree that was dropped by catkin, and source it.  It will
-	# set things like CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
-	if [ -f "/opt/ros/fuerte/setup.sh" ]; then . "/opt/ros/fuerte/setup.sh"; fi && \
-	dh_auto_build
-
-override_dh_auto_test:
-	# In case we're installing to a non-standard location, look for a setup.sh
-	# in the install tree that was dropped by catkin, and source it.  It will
-	# set things like CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
-	echo -- Running tests. Even if one of them fails the build is not canceled.
-	if [ -f "/opt/ros/fuerte/setup.sh" ]; then . "/opt/ros/fuerte/setup.sh"; fi && \
-	dh_auto_test || true
-
-override_dh_shlibdeps:
-	# In case we're installing to a non-standard location, look for a setup.sh
-	# in the install tree that was dropped by catkin, and source it.  It will
-	# set things like CMAKE_PREFIX_PATH, PKG_CONFIG_PATH, and PYTHONPATH.
-	if [ -f "/opt/ros/fuerte/setup.sh" ]; then . "/opt/ros/fuerte/setup.sh"; fi && \
-	dh_shlibdeps -l$(CURDIR)/debian/ros-fuerte-opencv2//opt/ros/fuerte/lib/
+#!/usr/bin/make -f
+# -*- makefile -*-
+# Sample debian/rules that uses debhelper.
+# This file was originally written by Joey Hess and Craig Small.
+# As a special exception, when this file is copied by dh-make into a
+# dh-make output file, you may use that output file without restriction.
+# This special exception was added by Craig Small in version 0.37 of dh-make.
+
+# Uncomment this to turn on verbose mode.
+export DH_VERBOSE=1
+export DH_OPTIONS=-v
+export CMAKE_PREFIX_PATH="/opt/ros/fuerte"
+CMAKE_FLAGS = \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DENABLE_SSE=ON \
+	-DENABLE_SSE2=ON \
+	-DENABLE_SSE3=ON \
+	-DENABLE_SSE42=OFF \
+	-DENABLE_SSE41=OFF \
+	-DUSE_FAST_MATH=OFF \
+	-DBUILD_EXAMPLES=OFF \
+	-DBUILD_DOCS=ON \
+	-DBUILD_TESTS=OFF \
+	-DBUILD_NEW_PYTHON_SUPPORT=ON \
+	-DWITH_V4L=ON \
+	-DWITH_CUDA=OFF \
+
+%:
+	dh  $@
+
+override_dh_auto_configure:
+	dh_auto_configure -Scmake -- \
+		-DCMAKE_INSTALL_PREFIX="/opt/ros/fuerte" \
+		-DCMAKE_PREFIX_PATH="/opt/ros/fuerte" \
+		-DCATKIN=YES \
+		$(CMAKE_FLAGS)
+
-- 
1.7.10.2 (Apple Git-33)

